src/api/app/models/user.rb

```diff
@@ -216,29 +216,31 @@ def self.find_with_credentials_via_ldap(login, password)
     user
   end
 
-  # The currently logged in user (might be nil). It's reset after
-  # every request and normally set during authentification
-  def self.current
-    Thread.current[:user]
-  end
-
-  def self.nobody
-    Thread.current[:nobody] ||= find_nobody!
-  end
-
   # Currently logged in user or nobody user if there is no user logged in.
   # Use this to check permissions, but don't treat it as logged in user. Check
   # is_nobody? on the returned object
-  def self.current_or_nobody
+  def self.possibly_nobody
     current || nobody
   end
 
-  def self.current=(user)
-    Thread.current[:user] = user
+  # Currently logged in user. Will thrown an exception if no user is logged in.
+  # So the controller needs to require login if using this (or models using it)
+  def self.required
+    raise ArgumentError, 'Requiring user, but found nobody' unless current && !current.is_nobody?
+    current
```

Comment:
Conceptually I would find it nicer if `required` only takes care of ensuring that there is a user set and setting the user and all that happens elsewhere.
So I could write in my code:

```ruby
def index
  User.required!
  render xml: ::Cloud::Backend::UploadJob.all(::User.current, format: :xml)
end
```

It might even make sense to call this before calling the action or methd. But this seems to require some more refactoring.

Reply:
I think this would also allow us to drop the `possibly_nobody` again.The `required!` method would take care of permitting access to nobody and `current` would take care of returning a user ('nobody' in case it's nil)

Reply:
I'm actually fine renaming possibly_nobody back to current - once we drop assigning nobody to current. 

I really would like to take this refactoring step by step, because a lot can break if you screw it. So first step: deprecate User.current and start to instrument code if requiring a user. And I can't continue this refactoring in a parallel branch as every code change needs to follow this new paradigm.

In a controller your code looks good, but in a partial it's already problematic. But possibly it's enough. Later on pundit can also set this required! thing

Reply:
That's why I didn't make it a change request. You mentioned that in the PR description that you plan a couple of more changes. It's just a bit hard to see where we (you actually:laughing:) are heading right now.
And I would like you to consider this early on:-)
But if the current `User.required` for a longer time I would have mixed feelings.

> In a controller your code looks good, but in a partial it's already problematic.

My thinking is that if we happen to require a non-nobody user in a partial, or any other method, we should ensure upfront that this is the case.
But it's true we might need to take some steps to get there.